src/pow.cpp

```diff
@@ -217,20 +213,31 @@ unsigned int DarkGravityWave(const CBlockIndex* pindexLast, const Consensus::Par
         pindexLastMatchingProof = pindexLast;
     }
 
+    bool fNewPoW = false;
     int64_t nPowSpacing;
     if (nPoWType & CBlockHeader::PROGPOW_BLOCK) {
         nPowSpacing = params.nProgPowTargetSpacing;
+        fNewPoW = true;
     } else if (nPoWType & CBlockHeader::RANDOMX_BLOCK) {
         nPowSpacing = params.nRandomXTargetSpacing;
+        fNewPoW = true;
     } else if (nPoWType & CBlockHeader::SHA256D_BLOCK) {
         nPowSpacing = params.nSha256DTargetSpacing;
+        fNewPoW = true;
     } else {
         nPowSpacing = params.nPowTargetSpacing;
     }
 
     int64_t nActualTimespan = pindexLastMatchingProof->GetBlockTime() - pindex->GetBlockTime();
     int64_t nTargetTimespan = params.nDgwPastBlocks * nPowSpacing;
 
+    // If we are calculating the diff for RandomX, Sha256, or ProgPow
+    // We want to take into account the latest block tip time into the calculation
+    int64_t nTipToLastMatchingProof = pindexLast->GetBlockTime() - pindexLastMatchingProof->GetBlockTime();
+    if (fNewPoW && nTipToLastMatchingProof > nPowSpacing) {
+        nActualTimespan += nTipToLastMatchingProof - nPowSpacing;
```

Comment:
took a minute but makes sense.  `-nPowSpacing` removes the need to add another nPowSpacing to the `nTargetTimespan`, and still covers if we're more than nPowSpacing overdue.